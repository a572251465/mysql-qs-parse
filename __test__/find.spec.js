import MysqlParse from '../src'
const connectionWay = {
  host: 'localhost',
  user: 'root',
  password: 'location@root',
  database: 'super-admin-system'
}

describe('测试查询函数find方法', () => {
  test('find 无条件查询多条数据', async () => {
    const db = new MysqlParse(connectionWay)
    db.on('error', (info) => {
      console.log(info)
    })
    await db.open()

    const res = await db.find(['name', 'introduce'], 'cuIndexMenu')

    db && db.release()

    expect(Array.isArray(res)).toBeTruthy()
    expect(res.length > 0).toBeTruthy()
    expect(res[0]).toMatchSnapshot({
      name: expect.any(String),
      introduce: expect.any(String)
    })
  })

  test('find 按照条件进行查询', async () => {
    const db = new MysqlParse(connectionWay)
    await db.open()

    const res = await db.find({
      fields: ['name', 'introduce', 'details'],
      tableName: 'cuIndexMenu',
      where: {
        type: 1
      }
    })
    db && db.release()

    expect(Array.isArray(res)).toBeTruthy()
    expect(res.length > 0).toBe(true)
    expect(res.length).toBe(2)
  })

  test('find 测试查询不到数据', async () => {
    const db = new MysqlParse(connectionWay)
    await db.open()

    const res = await db.find(['name'], 'cuIndexMenu', {
      type: 100
    })

    db && db.release()

    expect(res.length).toBe(0)
  })

  test('find limit 测试', async () => {
    const db = new MysqlParse(connectionWay)
    await db.open()

    const res = await db.find({
      fields: ['name'],
      tableName: 'cuIndexMenu',
      limit: {
        page: 1,
        limit: 2
      }
    })

    db && db.release()

    expect(res.length).toBe(2)
  })

  test('find order 排序测试', async () => {
    const db = new MysqlParse(connectionWay)
    db.on('error', (info) => {
      console.log(info)
    })
    await db.open()

    const res = await db.find({
      fields: ['type'],
      tableName: 'cuIndexMenu',
      order: {
        type: 'top'
      }
    })

    db && db.release()

    expect(+res[0]['type'] === 1 && +res[2]['type'] === 2 && +res[res.length - 1]['type'] === 6).toBeTruthy()
  })
})
