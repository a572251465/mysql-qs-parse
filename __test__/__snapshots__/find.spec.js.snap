// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试查询函数find方法 find 无条件查询多条数据 1`] = `
Object {
  "introduce": Any<String>,
  "name": Any<String>,
}
`;
